app/services/foreman_discovery/host_converter.rb

```diff
@@ -19,6 +20,41 @@ def self.to_managed(original_host, set_managed = true, set_build = true, added_a
     host
   end
 
+  def self.eventually_make_bond(host)
+    primary = host.primary_interface
+    return if primary.nil?
+    return if primary.type == 'Nic::Bond'
+
+    neighbors = ::ForemanDiscovery::LldpNeighbors
+      .from_facts(host.facts_hash)
+      .get_neighbors_by_interface(primary.identifier)
+
+    return if neighbors.nil?
+
+    ip = primary.ip
+    name = primary.name
+    primary.update(
+       :primary   => false,
+       :provision => false,
+       :managed   => false,
+       :name      => nil,
+       :ip        => nil
+    )
+
+    bond = Nic::Bond.create(
+        :identifier       => "bond0",
+        :attached_devices => neighbors,
+        :primary          => true,
+        :provision        => true,
+        :name             => name,
+        :ip               => ip,
+        :host             => host
+    )
+
+    bond.save!
```

Comment:
I'm wondering if it's a good idea to save the host here.
`ForemanDiscovery::HostConverter.to_managed` just converts the host object but does not save it so a user can still make changes to the host in the UI or even abort the operation. Same for the call to `update` above.

Maybe this should just update the models but not persist the changes to the database?
What do you think?

Reply:
I would prefer not to do so, but haven't been able to figure out a clean way. Problem seemed to be that at this time the primary_interface has already been stored. Removing the primary flag and creating another primary interface always led to a conflict. It behaved like a check that verified the database first without taking the in-memory changes into account.

Reply:
You could give it a try, but as far as I remember when starting with an empty database you'll see a single (managed, primary) interface in the DB as soon as a host with (multiple) interfaces got "discovered".

Reply:
Hello, thanks for your contribution! Very nicely done.

Now, what @timogoebel mentioned is a real issue - this method was designed specifically in a way that it does not store anything in database, it is called when Edit Host form is being rendered to convert host to Host::Managed so the form from Foreman Core can be reused. We wanted to avoid creating New Host form, it's the most complicated one in the whole project.

Reply:
You could try to actually update the host object with nested attributes. Just a thought.